### DISCUSSION TITLE
[Bug]: Pimcore does not install "no such file or directory"

### DISCUSSION THREAD

**Discussion by Chris-Gett** [pimcore/pimcore#9953] [Q&A / Support] [answered]
### Expected behavior

After running composer and switching to my project folder the command [ ./my-project/vendor/bin/pimcore-install ] generates the error message "no such file or directory" 

### Actual behavior

**these are the messages while composer is running :**
(I have anonymised the username)

[user**** ~]$ COMPOSER_MEMORY_LIMIT=-1 composer create-project pimcore/skeleton pimcore-gett-services
Creating a "pimcore/skeleton" project at "./pimcore-gett-services"
Installing pimcore/skeleton (v10.0.9)
  - Installing pimcore/skeleton (v10.0.9): Extracting archive
Created project in /home/gettservices/pimcore-gett-services
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - endroid/qr-code is locked to version 4.2.1 and an update of this package was not requested.
    - endroid/qr-code 4.2.1 requires php ^7.3||^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 2
    - http-interop/http-factory-guzzle is locked to version 1.2.0 and an update of this package was not requested.
    - http-interop/http-factory-guzzle 1.2.0 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 3
    - knplabs/knp-components is locked to version v3.1.0 and an update of this package was not requested.
    - knplabs/knp-components v3.1.0 requires php ^7.3 || ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 4
    - knplabs/knp-paginator-bundle is locked to version v5.6.0 and an update of this package was not requested.
    - knplabs/knp-paginator-bundle v5.6.0 requires php ^7.3 || ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 5
    - laminas/laminas-code is locked to version 4.4.2 and an update of this package was not requested.
    - laminas/laminas-code 4.4.2 requires php ^7.4 || ~8.0.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 6
    - lcobucci/clock is locked to version 2.0.0 and an update of this package was not requested.
    - lcobucci/clock 2.0.0 requires php ^7.4 || ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 7
    - lcobucci/jwt is locked to version 4.1.4 and an update of this package was not requested.
    - lcobucci/jwt 4.1.4 requires ext-sodium * -> it is missing from your system. Install or enable PHP's sodium extension.
  Problem 8
    - league/csv is locked to version 9.7.1 and an update of this package was not requested.
    - league/csv 9.7.1 requires php ^7.3 || ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 9
    - myclabs/php-enum is locked to version 1.8.3 and an update of this package was not requested.
    - myclabs/php-enum 1.8.3 requires php ^7.3 || ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 10
    - ocramius/package-versions is locked to version 2.3.0 and an update of this package was not requested.
    - ocramius/package-versions 2.3.0 requires php ~8.0.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 11
    - onnov/detect-encoding is locked to version v2.0.0 and an update of this package was not requested.
    - onnov/detect-encoding v2.0.0 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 12
    - pimcore/pimcore is locked to version v10.1.0 and an update of this package was not requested.
    - pimcore/pimcore v10.1.0 requires php ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 13
    - psr/link is locked to version 1.1.1 and an update of this package was not requested.
    - psr/link 1.1.1 requires php >=8.0.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 14
    - webmozarts/console-parallelization is locked to version 1.2.4 and an update of this package was not requested.
    - webmozarts/console-parallelization 1.2.4 requires php ^7.3 || ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 15
    - codeception/module-symfony is locked to version 1.6.0 and an update of this package was not requested.
    - codeception/module-symfony 1.6.0 requires php ^7.3 | ^8.0 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 16
    - phpunit/php-code-coverage is locked to version 9.2.6 and an update of this package was not requested.
    - phpunit/php-code-coverage 9.2.6 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 17
    - phpunit/php-file-iterator is locked to version 3.0.5 and an update of this package was not requested.
    - phpunit/php-file-iterator 3.0.5 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 18
    - phpunit/php-invoker is locked to version 3.1.1 and an update of this package was not requested.
    - phpunit/php-invoker 3.1.1 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 19
    - phpunit/php-text-template is locked to version 2.0.4 and an update of this package was not requested.
    - phpunit/php-text-template 2.0.4 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 20
    - phpunit/php-timer is locked to version 5.0.3 and an update of this package was not requested.
    - phpunit/php-timer 5.0.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 21
    - phpunit/phpunit is locked to version 9.5.8 and an update of this package was not requested.
    - phpunit/phpunit 9.5.8 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 22
    - sebastian/cli-parser is locked to version 1.0.1 and an update of this package was not requested.
    - sebastian/cli-parser 1.0.1 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 23
    - sebastian/code-unit is locked to version 1.0.8 and an update of this package was not requested.
    - sebastian/code-unit 1.0.8 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 24
    - sebastian/code-unit-reverse-lookup is locked to version 2.0.3 and an update of this package was not requested.
    - sebastian/code-unit-reverse-lookup 2.0.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 25
    - sebastian/comparator is locked to version 4.0.6 and an update of this package was not requested.
    - sebastian/comparator 4.0.6 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 26
    - sebastian/complexity is locked to version 2.0.2 and an update of this package was not requested.
    - sebastian/complexity 2.0.2 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 27
    - sebastian/diff is locked to version 4.0.4 and an update of this package was not requested.
    - sebastian/diff 4.0.4 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 28
    - sebastian/environment is locked to version 5.1.3 and an update of this package was not requested.
    - sebastian/environment 5.1.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 29
    - sebastian/exporter is locked to version 4.0.3 and an update of this package was not requested.
    - sebastian/exporter 4.0.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 30
    - sebastian/global-state is locked to version 5.0.3 and an update of this package was not requested.
    - sebastian/global-state 5.0.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 31
    - sebastian/lines-of-code is locked to version 1.0.3 and an update of this package was not requested.
    - sebastian/lines-of-code 1.0.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 32
    - sebastian/object-enumerator is locked to version 4.0.4 and an update of this package was not requested.
    - sebastian/object-enumerator 4.0.4 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 33
    - sebastian/object-reflector is locked to version 2.0.4 and an update of this package was not requested.
    - sebastian/object-reflector 2.0.4 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 34
    - sebastian/recursion-context is locked to version 4.0.4 and an update of this package was not requested.
    - sebastian/recursion-context 4.0.4 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 35
    - sebastian/resource-operations is locked to version 3.0.3 and an update of this package was not requested.
    - sebastian/resource-operations 3.0.3 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 36
    - sebastian/type is locked to version 2.3.4 and an update of this package was not requested.
    - sebastian/type 2.3.4 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 37
    - sebastian/version is locked to version 3.0.2 and an update of this package was not requested.
    - sebastian/version 3.0.2 requires php >=7.3 -> your php version (7.2.34) does not satisfy that requirement.
  Problem 38
    - ocramius/package-versions 2.3.0 requires php ~8.0.0 -> your php version (7.2.34) does not satisfy that requirement.
    - symfony/proxy-manager-bridge v5.3.4 requires composer/package-versions-deprecated ^1.8 -> satisfiable by ocramius/package-versions[2.3.0].
    - symfony/proxy-manager-bridge is locked to version v5.3.4 and an update of this package was not requested.

To enable extensions, verify that they are enabled in your .ini files:
    - /opt/cpanel/ea-php72/root/etc/php.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/01-ioncube.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/10-opcache.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-bcmath.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-calendar.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-ctype.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-curl.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-dom.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-exif.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-fileinfo.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-ftp.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-gd.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-iconv.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-imap.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-intl.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-json.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-mbstring.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-mysqlnd.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-pdo.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-phar.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-posix.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-simplexml.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-soap.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-sockets.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-sqlite3.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-tokenizer.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-xml.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-xmlwriter.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-xsl.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/20-zip.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/30-mysqli.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/30-pdo_mysql.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/30-pdo_sqlite.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/30-wddx.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/30-xmlreader.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/zz-rackspeed.ini
    - /opt/cpanel/ea-php72/root/etc/php.d/zzzzzzz-pecl.ini
You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
[user**** ~]$ cd pimcore-gett-services/
[user**** pimcore-gett-services]$ ls -la
total 684
drwxrwxr-x  9 gettservices gettservices   4096 Aug  4 11:59 .
drwx--x--x 22 gettservices gettservices   4096 Aug  5 16:19 ..
drwxrwxr-x  2 gettservices gettservices   4096 Aug  4 11:59 bin
-rw-rw-r--  1 gettservices gettservices   1093 Aug  4 11:59 composer.json
-rw-rw-r--  1 gettservices gettservices 591174 Aug  4 11:59 composer.lock
drwxrwxr-x  6 gettservices gettservices   4096 Aug  4 11:59 config
-rw-rw-r--  1 gettservices gettservices    985 Aug  4 11:59 docker-compose.yml
-rw-rw-r--  1 gettservices gettservices    269 Aug  4 11:59 .editorconfig
-rw-rw-r--  1 gettservices gettservices    961 Aug  4 11:59 .env
-rw-rw-r--  1 gettservices gettservices      9 Aug  4 11:59 .gitattributes
-rw-rw-r--  1 gettservices gettservices    872 Aug  4 11:59 .gitignore
-rw-rw-r--  1 gettservices gettservices  35147 Aug  4 11:59 gpl-3.0.txt
-rw-rw-r--  1 gettservices gettservices   2978 Aug  4 11:59 .php_cs.dist
drwxrwxr-x  2 gettservices gettservices   4096 Aug  4 11:59 public
-rw-rw-r--  1 gettservices gettservices   1161 Aug  4 11:59 README.md
drwxrwxr-x  4 gettservices gettservices   4096 Aug  4 11:59 src
drwxrwxr-x  4 gettservices gettservices   4096 Aug  4 11:59 templates
drwxrwxr-x  2 gettservices gettservices   4096 Aug  4 11:59 translations
drwxrwxr-x  3 gettservices gettservices   4096 Aug  4 11:59 var
[user**** pimcore-gett-services]$ ./vendor/bin/pimcore-install
-bash: ./vendor/bin/pimcore-install: No such file or directory
[user**** pimcore-gett-services]$

### Steps to reproduce

i just followed the official installation manual for both packages (skeleton and demo) the error occurs on both installations

**Comment by brusch** [comment, marked as answer]
Pimcore required PHP 8 ... you're using 7.2.34 ...

**Reply by Chris-Gett** [reply]
Pimcore required PHP 8 ... you're using 7.2.34 ...

_Originally posted by @brusch in https://github.com/pimcore/pimcore/issues/9952#issuecomment-893518350_


PHP 8 is already running...

$ php -v
PHP 8.0.8 (cli) (built: Jul 13 2021 20:27:50) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.8, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.8, Copyright (c), by Zend Technologies
[user*** pimcore-gett-services]$
![2021-08-05 16_48_16-Greenshot](https://user-images.githubusercontent.com/73822840/128372590-ba0749bf-9f3e-4a47-ae7b-34b0f3dcc4ac.png)

**Reply by alexej-d** [reply]
If you execute `php -v` in your console, what does it say? To me it looks like you are only using PHP8 for the frontend, not for the shell…

**Reply by Chris-Gett** [reply]
I´m Sorry, you´re totally right. The wrong PHP version was installed in the shell...  I had not even thought about it, because it is actually a server optimized for PIMCORE (according to the information from the hoster)...

**Reply by alexej-d** [reply]
@Chris-Gett happens to the best 😊 Gutes Gelingen!
